docs/architecture/adr-074-timeout-params.md

```diff
@@ -0,0 +1,224 @@
+# ADR 74: Migrate Timeout Parameters to Consensus Parameters
+
+## Changelog
+
+- 03-Jan-2022: Initial draft (@williambanfield)
+
+## Status
+
+Proposed
+
+## Context
+
+### Background
+
+Tendermint's consensus timeout parameters are currently configured locally by each validator
+in the validator's [config.toml][config-toml].
+This means that the validators on a Tendermint network may have different timeouts
+from each other. There is no reason for validators on the same network to configure
+different timeout values. Proper functioning of the Tendermint consensus algorithm
+relies on these parameters being uniform across validators.
+
+The configurable values are as follows:
+
+* `TimeoutPropose`
+	* How long the consensus algorithm waits for a proposal block before issuing a prevote.
+	* If no prevote arrives by `TimeoutPropose`, then the consensus algorithm will issue a nil prevote.
+* `TimeoutProposeDelta`
+	* How much the `TimeoutPropose` grows each round.
+* `TimeoutPrevote`
+	* How long the consensus algorithm waits after receiving +2/3 prevotes with
+	no quorum for a value before issuing a precommit for nil.
+	(See the [arXiv paper][arxiv-paper], Algorithm 1, Line 34)
+* `TimeoutPrevoteDelta`
+	* How much the `TimeoutPrevote` increases with each round.
+* `TimeoutPrecommit`
+	* How long the consensus algorithm waits after receiving +2/3 precommits that
+	do not have a quorum for a value before entering the next round. 
+	(See the [arXiv paper][arxiv-paper], Algorithm 1, Line 47)
+* `TimeoutPrecommitDelta`
+	* How much the `TimeoutPrecommit` increases with each round.
+* `TimeoutCommit`
```

Comment:
So just to confirm, this timer starts after we call the `Commit` ABCI method right? Not immediately after receiving 2/3+ precommits

Reply:
Double checking this, I'm seeing that we calculate the `StartTime` for the next height using the `CommitTime` of the previous height.

`CommitTime` is set to the be the time that the validator received +2/3 precommits:
https://github.com/tendermint/tendermint/blob/7cdf560173dee6773b80d1c574a06489d4c394fe/internal/consensus/state.go#L1598

In effect, this means that `TimeoutCommit` starts when +2/3 precommits are received, not after the application has finished the computation for the commit. 

What tripped me up is that we set the variable after the commit completes, but it used the previously captured `CommitTime` value.

Reply:
Okay. This seems reasonable to me. Thanks for double checking